launcher/internal/gpu/driverinstaller.go

```diff
@@ -167,3 +198,82 @@ func remountAsExecutable(dir string) error {
 	}
 	return nil
 }
+
+func verifyDriverInstallation() error {
+	// Run nvidia-smi to check whether nvidia GPU driver is installed.
+	if err := exec.Command("nvidia-smi").Run(); err != nil {
+		return fmt.Errorf("failed to verify gpu driver installation : %v", err)
+	}
+	return nil
+}
+
+func setGPUStateToReady() error {
+	// Run nvidia-smi conf-compute command to set GPU state to READY.
+	if err := exec.Command("nvidia-smi", "conf-compute", "-srs", "1").Run(); err != nil {
+		return fmt.Errorf("failed to set the gpu state to ready: %v", err)
+	}
+	return nil
+}
+
+func isGPUCCModeEnabled(logger logging.Logger, gpuType deviceinfo.GPUType) (bool, error) {
+	// The nvidia-smi conf-compute command fails for GPU which doesn't support confidential computing.
+	// This check would bypass nvidia-smi conf-compute command for GPU not having confidential compute support.
+	if !isCCSupportedGpu(gpuType) {
+		logger.Info("Confidential Computing is not supported for GPU type : %s", gpuType.String())
+		return false, nil
+	}
+	// Run nvidia-smi conf-compute command to check if confidential compute mode is ON.
+	ccModeOutput, err := exec.Command("nvidia-smi", "conf-compute", "-f").Output()
+	if err != nil {
+		return false, err
+	}
+	return strings.Contains(string(ccModeOutput), "CC status: ON"), nil
+}
+
+func isCCSupportedGpu(gpuType deviceinfo.GPUType) bool {
+	switch gpuType {
+	case deviceinfo.H100:
+		return true
+	default:
+		return false
+	}
+}
+
+func launchNvidiaPersistencedProcess(logger logging.Logger) error {
+	newPathEnv := fmt.Sprintf("%s/bin:%s", InstallationHostDir, os.Getenv("PATH"))
+	os.Setenv("PATH", newPathEnv)
+	logger.Info("Starting nvidia-persistenced process")
+	if err := exec.Command("nvidia-persistenced").Run(); err != nil {
```

Comment:
Ideally we should use the systemd to manage the lifecycle of this process, but I guess for now it is fine